src/libs/signer.service.ts

```diff
@@ -25,34 +32,77 @@ export function getSigner() {
   return signer;
 }
 
-export async function signTransaction(signingRequest: SigningRequest) {
+export async function signTransaction(
+  signingRequest: SigningRequest
+): Promise<SigningResponse> {
   const signer = getSigner();
   const provider = signer.provider;
 
   if (!provider) throw new Error("RPC provider not specified");
 
+  let signerBalanceInWei: ethers.BigNumber;
+  try {
+    signerBalanceInWei = await signer.getBalance();
+  } catch (error) {
+    logger.error("❌ Unable to get signing key balance.");
```

Comment:
im not sure these signing key balance checks are really necessary. It adds calls that are not really relevant to show case how the relayer works. Its more like implementation error handling which is beyond the scope of this priject.

It will be obvious from the error message when sending the tx that the key doesnt have enought balance if it happens